File: hackerrank/easy/0016-counting-valleys/python/solution.py
## solution.py

def counting_valleys(steps: int, path: str) -> int:
    level = 0
    valleys = 0
    for step in path:
        if step == "U":
            level += 1
            if level == 0:
                valleys += 1
        else:
            level -= 1
    return valleys


# --- demo run ---
if __name__ == "__main__":
    print(counting_valleys(8, "UDDDUDUU"))  # 1

File: hackerrank/easy/0016-counting-valleys/js/solution.mjs
// solution.mjs

export function countingValleys(steps, path) {
  let level = 0;
  let valleys = 0;
  for (const step of path) {
    if (step === "U") {
      level++;
      if (level === 0) valleys++;
    } else {
      level--;
    }
  }
  return valleys;
}

// --- demo run ---
console.log(countingValleys(8, "UDDDUDUU")); // 1